flake.nix

```diff
@@ -5,27 +5,76 @@
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   };
 
-  outputs = { self, flake-utils, naersk, nixpkgs }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
+  outputs = {
+    self,
+    flake-utils,
+    naersk,
+    nixpkgs,
+  }:
+    flake-utils.lib.eachDefaultSystem (
+      system: let
         pkgs = (import nixpkgs) {
           inherit system;
         };
+        inherit (pkgs) lib;
+
         naersk' = pkgs.callPackage naersk {};
-        nativeBuildInputs = with pkgs; [ pkg-config ];
-      in rec {
-        packages.default = naersk'.buildPackage {
-          inherit nativeBuildInputs;
-          src = ./.;
-          buildInputs = with pkgs; [ openssl ] ++ (pkgs.lib.lists.optionals (stdenv.isDarwin) [ darwin.apple_sdk.frameworks.SystemConfiguration ]);
+        nativeBuildInputs = with pkgs; [pkg-config];
+
+        darwin_frameworks = with pkgs.darwin.apple_sdk.frameworks; [
+          Security
+          SystemConfiguration
+        ];
+
+        dev_tools = with pkgs; [
+          cargo
+          rustc
+        ];
+
+        dependencies = with pkgs;
+          [
+            libgit2
+            openssl
+          ]
+          ++ lib.lists.optionals stdenv.isDarwin darwin_frameworks;
+
+        env_vars = {
+          LIBGIT2_NO_VENDOR = 1;
+          OPENSSL_NO_VENDOR = 1;
         };
-        devShells.default = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; nativeBuildInputs ++ [
-            cargo
-            rustc
-          ];
+      in rec {
+        # NB: if this does not build and you need to modify the file,
+        #     please ensure you also make the corresponding changes in the devshell
+        packages.default = naersk'.buildPackage ({
+            inherit nativeBuildInputs;
+            src = ./.;
+            buildInputs = dev_tools ++ dependencies;
+          }
+          // env_vars);
+
+        devShells.default = pkgs.mkShell ({
+            buildInputs = nativeBuildInputs ++ dev_tools ++ dependencies;
+          }
+          // env_vars);
+
+        formatter = pkgs.alejandra;
+
+        checks = {
+          builds = packages.default;
+          nix-files-are-formatted = pkgs.stdenvNoCC.mkDerivation {
+            name = "fmt-check";
+            dontBuild = true;
+            src = ./.;
+            doCheck = true;
+            nativeBuildInputs = with pkgs; [alejandra];
+            checkPhase = ''
+              alejandra -c .
```

Comment:
I think internally we're trying to use nixfmt and not alejandra. it'd be good to at least be consistent.

Reply:
also, https://github.com/NixOS/rfcs/pull/166

internal rationale for posterity:

```
The three main options are:
[nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt): this is the most "official" of the bunch, maintained in the nix-community GitHub org and designed to format nixpkgs. It has a weird design choice though of "only expand, never collapse" which tends to result in long files, with short line lengths, during my testing.
[nixfmt](https://github.com/serokell/nixfmt): I tried this out after not liking the "only expand, never collapse" of nixpkgs-fmt. It was fine: fast, installed easily (with Nix, of course) and integrated with my editor and pre-commit easily. I've been using it on side projects a few weeks without issues.
[alejandra](https://github.com/kamadorueda/alejandra): less actively maintained, and resulted in the largest diffs when I ran it on a subset of nixpkgs, which suggests its preferred style is divergent from community norms.
```

Reply:
Happy to do so. It's worth mentioning that nixpkgs will use both, and alejandra is a direct response to nixfmt breaking some nixpkgs builds.

Reply:
If RFC 166 will be accepted then Nixpkgs will not use both. Alejandra raised the bar when it was created, but nowadays Nixfmt fulfills pretty much the same correctness guarantees. (Though be sure to use the `nixfmt-rfc-style` package or the RFC development branch.)